File: Summation_Nonfactors/Sum.ts
/*
    Author : Pakshal Jain
    Date : 05/10/2022
    Program : To Calculate Summation of Nonfactors of Entered Number
*/

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Function To Calulate summation
function Summation(no : number) 
{
    var i : number = 0;
    var sum : number = 0;

    //Logic
    for(i = 1;i < no;i++)
    {
        if(no % i != 0)
        {
            sum = sum + i;
        }
    }

    return sum;
}

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

function main() 
{
    var no : number = 12;
    var iret : number = 0;

    console.log("Jay Ganesh....");
    
    //Call To Summation Function
    iret = Summation(no);

    console.log("Summation of Non Factors of Entered Number is : %d",iret);
}

//Call To  Main Function
main();

File: Summation_Nonfactors/Sum.go
/*
	Author : Pakshal Jain
	Date : 05/10/2022
	Program : To Print Summation of Nonfactors of Entered Number
*/

//Required Libraries
package main
import "fmt"

//Function To Calculate Summation of Nonfactors of Number
func Summation(no int) int {
	var i int = 0;
	var sum int = 0;

	//Logic
	for i = 1;i < no;i++ {
		if no % i != 0 {
			sum = sum + i;
		}
	}

	return sum;
}

//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

func main() {
	var No int = 0;
	var iret int = 0;

	fmt.Println("Jay Ganesh....");
	
	fmt.Println("Enter One Number..");
	fmt.Scanf("%d",&No);
	
	//Call To Summation Function
	iret = Summation(No);
	fmt.Printf("Summation of Nonfactors of Entered Numbe is : %d",iret);
}
